src/main/java/com/teragrep/lsh_01/conversion/RegexConversion.java

```diff
@@ -0,0 +1,74 @@
+/*
+  logstash-http-input to syslog bridge
+  Copyright 2024 Suomen Kanuuna Oy
+
+  Derivative Work of Elasticsearch
+  Copyright 2012-2015 Elasticsearch
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+package com.teragrep.lsh_01.conversion;
+
+import com.teragrep.lsh_01.authentication.Subject;
+
+import java.util.Map;
+import java.util.regex.Pattern;
+
+/**
+ * Decorator for IMessageHandler that splits messages based on given regex.
+ */
+public class RegexConversion implements IMessageHandler {
+
+    private final IMessageHandler conversion;
+    private final String regex;
+
+    public RegexConversion(IMessageHandler conversion, String regex) {
+        this.conversion = conversion;
+        this.regex = regex;
+    }
+
+    @Override
+    public boolean onNewMessage(Subject subject, Map<String, String> headers, String body) {
+        Pattern splitPattern = Pattern.compile(regex);
```

Comment:
here pattern gets compiled for each `onNewMessage()` call, perhaps change primary constructor to take Pattern as parameter and secondary constructor can have the String regex

this way the pattern is only compiled once for the object

Reply:
Nice catch, made a secondary ctor for this.